File: crossecore-java/src/java_/ManifestBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class ManifestBase 
extends org.eclipse.emf.ecore.impl.MinimalEObjectImpl.Container implements Manifest
{
	private com.crossecore.ocl.Set<ManifestAttribute> _mainAttributes;
	
	public com.crossecore.ocl.Set<ManifestAttribute> getMainAttributes()
	{
		if(_mainAttributes==null){
			_mainAttributes = new com.crossecore.ocl.Set<ManifestAttribute>(ManifestAttribute.class, this, Java_PackageImpl.MANIFEST__MAIN_ATTRIBUTES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MANIFEST__MAIN_ATTRIBUTES);
		}
		return _mainAttributes;
	
	}
	private com.crossecore.ocl.Set<ManifestEntry> _entryAttributes;
	
	public com.crossecore.ocl.Set<ManifestEntry> getEntryAttributes()
	{
		if(_entryAttributes==null){
			_entryAttributes = new com.crossecore.ocl.Set<ManifestEntry>(ManifestEntry.class, this, Java_PackageImpl.MANIFEST__ENTRY_ATTRIBUTES, EOPPOSITE_FEATURE_BASE - Java_PackageImpl.MANIFEST__ENTRY_ATTRIBUTES);
		}
		return _entryAttributes;
	
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.MANIFEST;
	}
	
	
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST__MAIN_ATTRIBUTES:
				return getMainAttributes();
			case Java_PackageImpl.MANIFEST__ENTRY_ATTRIBUTES:
				return getEntryAttributes();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST__MAIN_ATTRIBUTES:
				getMainAttributes().clear();
				getMainAttributes().addAll((java.util.Collection<? extends ManifestAttribute>) newValue);
				return;
			case Java_PackageImpl.MANIFEST__ENTRY_ATTRIBUTES:
				getEntryAttributes().clear();
				getEntryAttributes().addAll((java.util.Collection<? extends ManifestEntry>) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.MANIFEST__MAIN_ATTRIBUTES:
				return _mainAttributes != null && !_mainAttributes.isEmpty();
			case Java_PackageImpl.MANIFEST__ENTRY_ATTRIBUTES:
				return _entryAttributes != null && !_entryAttributes.isEmpty();
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/ManifestBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {AbstractCollection} from "ecore/AbstractCollection";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {ManifestEntry} from "./ManifestEntry";
import {ManifestAttribute} from "./ManifestAttribute";
import {Manifest} from "./Manifest";
		
			export class ManifestBase
			extends BasicEObjectImpl
			implements Manifest
			{
				private _mainAttributes:Set<ManifestAttribute> = null;
				
				get mainAttributes():Set<ManifestAttribute>{
					if(this._mainAttributes===null){
						this._mainAttributes = new Set<ManifestAttribute>(this, Java_PackageLiterals.MANIFEST_MAINATTRIBUTES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MANIFEST_MAINATTRIBUTES);
							
					}
					return this._mainAttributes;
					
				}
				
				
				private _entryAttributes:Set<ManifestEntry> = null;
				
				get entryAttributes():Set<ManifestEntry>{
					if(this._entryAttributes===null){
						this._entryAttributes = new Set<ManifestEntry>(this, Java_PackageLiterals.MANIFEST_ENTRYATTRIBUTES, BasicEObjectImpl.EOPPOSITE_FEATURE_BASE - Java_PackageLiterals.MANIFEST_ENTRYATTRIBUTES);
							
					}
					return this._entryAttributes;
					
				}
				
				

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return ManifestBase.eStaticClass;
				}
			
			
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.MANIFEST_MAINATTRIBUTES:
							return this.mainAttributes;
						case Java_PackageLiterals.MANIFEST_ENTRYATTRIBUTES:
							return this.entryAttributes;
					}
					//return this.eGetFromBasicEObjectImpl(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.MANIFEST_MAINATTRIBUTES:
							this.mainAttributes.clear();
							this.mainAttributes.addAll(newValue);
							return;
						case Java_PackageLiterals.MANIFEST_ENTRYATTRIBUTES:
							this.entryAttributes.clear();
							this.entryAttributes.addAll(newValue);
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
